main.go

```diff
@@ -42,88 +42,100 @@ func main() {
 //	number of line deleted
 //	list of function calls seen in the diffs and their number of calls
 func computeDiff() *diffResult {
+	// Index all files in the diffs folder
 	rootFolder := "./diffs/"
 	diffFiles, err := ioutil.ReadDir(rootFolder)
 	if err != nil {
 		log.Fatal(err)
 	}
 
+	// Start a routine for each file
 	resChannel := make(chan diffResult)
-
 	expectedResults := 0
 	for _, file := range diffFiles {
 		go parseFileDiff(rootFolder+file.Name(), resChannel)
 		expectedResults++
 	}
 
-	res := diffResult{
+	diffResult := diffResult{
 		functionCalls: make(map[string]int),
 	}
 
+	// Get results from the communication channel
 	for i := 0; i < expectedResults; i++ {
 		routineRes := <-resChannel
 
-		res.regions += routineRes.regions
-		res.lineAdded += routineRes.lineAdded
-		res.lineDeleted += routineRes.lineDeleted
-		res.files = append(res.files, routineRes.files...)
+		diffResult.regions += routineRes.regions
+		diffResult.lineAdded += routineRes.lineAdded
+		diffResult.lineDeleted += routineRes.lineDeleted
+		diffResult.files = append(diffResult.files, routineRes.files...)
 
 		for key, value := range routineRes.functionCalls {
-			res.functionCalls[key] += value
+			diffResult.functionCalls[key] += value
 		}
 	}
 
-	return &res
+	return &diffResult
 }
 
+// Parse the content of a diff file and send the result through the channel
 func parseFileDiff(filename string, resChannel chan diffResult) {
-	file, err := os.Open(filename)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	scanner := bufio.NewScanner(file)
-
+	// Define all the regex that we need to parse the Diffs
+	var regionPrefix = "^@@"
+	var diffCallPrefix = "diff --git "
+	var diffCallRegex = regexp.MustCompile(regexp.QuoteMeta(diffCallPrefix))
+	var deletedPrefix = "^" + regexp.QuoteMeta("- ")
+	var addedPrefix = "^" + regexp.QuoteMeta("+ ")
```

Comment:
I'm not sure I understand why you have an empty character after the `+` and `-`. 

How would you detect something like:

`+void intel_bios_cleanup(struct drm_i915_private *dev_priv);`